examples/post-processing/rs-post-processing.cpp

```diff
@@ -0,0 +1,411 @@
+    double yaw, pitch; float offset_x, offset_y; texture tex;
+};
+
+// Struct of parameters controled by the user
+struct params {
+    bool do_decimate, do_spatial, do_temporal, do_disparity;
+    int decimate_magnitude, temporal_magnitude, spatial_magnitude, temp_smooth_delta, spat_smooth_delta;
+    float temp_smooth_alpha, spat_smooth_alpha;
+};
+
+struct filter_ranges {
+    rs2::option_range dec_magnitude_range, spat_magnitude_range, spat_smooth_alpha_range,
+        spat_smooth_delta_range, temp_smooth_alpha_range, temp_smooth_delta_range, temp_magnitude_range;
+};
+
+
+// helper functions for rendering the ui
+void render_slider_int(float3 location, char* name, char* label, int* slider_param,
+    int min, int max, char* description);
+void render_slider_float(float3 location, char* name, char* label, float* slider_param,
+    float min, float max, char* description);
+void render_checkbox(float2 location, bool* checkbox_param, char* label);
+void render_ui(float w, float h, params& params, filter_ranges& ranges);
+
+// helper functions to handle filter options
+void get_filter_ranges(filter_ranges& ranges, rs2::decimation_filter& dec_filter,
+    rs2::spatial_filter& spat_filter, rs2::temporal_filter& temp_filter);
+void set_defaults(params& params, filter_ranges& ranges);
+void set_filter_options(params& params, rs2::decimation_filter& dec_filter,
+    rs2::spatial_filter& spat_filter, rs2::temporal_filter& temp_filter);
+
+// A function that draws the 3D-display
+void draw_pointcloud(const rect& viewer_rect, state& app_state, rs2::points& points);
+
+int main(int argc, char * argv[]) try
+{
+    // Create a simple OpenGL window for rendering:
+    window app(1280, 720, "RealSense Post Processing Example");
+    ImGui_ImplGlfw_Init(app, false);
+
+    // Taking dimensions of the window for rendering purposes
+    float w = static_cast<float>(app.width());
+    float h = static_cast<float>(app.height());
+
+    // Construct objects to manage view state
+    state app_state;
+    state app_state_filtered;
+
+    /* Declare pointcloud objects, for calculating pointclouds and texture mappings
+    one is for the original frames and the second is for the filtered ones */
+    rs2::pointcloud pc;
+    rs2::pointcloud pc_filtered;
+
+    // Declare depth colorizer for pretty visualization of depth data
+    rs2::colorizer color_map;
+
+    /* Declare objects that will hold the calculated pointclouds, one is for the
+    original and one for filtered */
+    rs2::points depth_points;
+    rs2::points filtered_points;
+
+    // Declare RealSense pipeline, encapsulating the actual device and sensors
+    rs2::pipeline pipe;
+    // Start streaming with default recommended configuration
+    pipe.start();
+
+    /* save the the time of current frame's arrival in last_time
+    curr will hold the next frame's arrival time */
+    long long last_time = std::chrono::high_resolution_clock::now().time_since_epoch().count();
```

Comment:
When dealing with time, stick to using `std::chrono::duration` and `std::chrono::time_point`.
Just save `auto now = std::chrono::high_resolution_clock::now()` and declare `curr` when you first use it (not need to declare it here)
Later on:
```cpp
const std::chrono::milliseconds rotation_delta(10);
auto curr = std::chrono::high_resolution_clock::now();
if(curr - now > rotation_delta)
{
    //...
}
```